hls-plugin-api/src/Ide/Types.hs

```diff
@@ -431,304 +414,332 @@ instance PluginRequestMethod TextDocumentCodeAction where
         = x
         | otherwise = InL cmd
         where
-          cmd = mkLspCommand "hls" "fallbackCodeAction" (action ^. title) (Just cmdParams)
-          cmdParams = [toJSON (FallbackCodeActionParams (action ^. edit) (action ^. command))]
+          cmd = mkLspCommand "hls" "fallbackCodeAction" (action ^. L.title) (Just cmdParams)
+          cmdParams = [toJSON (FallbackCodeActionParams (action ^. L.edit) (action ^. L.command))]
 
       wasRequested :: (Command |? CodeAction) -> Bool
       wasRequested (InL _) = True
       wasRequested (InR ca)
         | Nothing <- _only context = True
-        | Just (List allowed) <- _only context
+        | Just allowed <- _only context
         -- See https://github.com/microsoft/language-server-protocol/issues/970
         -- This is somewhat vague, but due to the hierarchical nature of action kinds, we
         -- should check whether the requested kind is a *prefix* of the action kind.
         -- That means, for example, we will return actions with kinds `quickfix.import` and
         -- `quickfix.somethingElse` if the requested kind is `quickfix`.
-        , Just caKind <- ca ^. kind = any (\k -> k `codeActionKindSubsumes` caKind) allowed
+        , Just caKind <- ca ^. L.kind = any (\k -> k `codeActionKindSubsumes` caKind) allowed
         | otherwise = False
 
-instance PluginMethod Request TextDocumentDefinition where
+      -- Copied form lsp-types 1.6 to get compilation working. May make more
```

Comment:
yes, we should do this, TODO MPJ

Reply:
The reason why I didn't incude it in the helper instances and functions pr is because it is only ever used this once.

Reply:
yeah, but logically it's part of LSP and something other people might want